### DISCUSSION TITLE
MorphMany comments resource broken with assets from >v4.5.2

### DISCUSSION THREAD

**Discussion by dmitrijsmihailovs** [laravel/nova-issues#4784] [Q&A] [answered]
- Laravel Version: 9.28
- Nova Version: 4.13
- PHP Version: 8.1
- Database Driver & Version: Mysql 8
- Operating System and Version: MacOS 12.5.1
- Browser type and version: Chrome 104
- Reproduction Repository: 

### Description:

We have a Comment resource that is used in other resources with morph relationship with fields
```
public function fields(NovaRequest $request): array
    {
        return [
            ID::make(),
            MorphTo::make(__('comment.type'), 'commentable')->types([
                // Resources
            ])->hide(),
            BelongsTo::make(__('comment.author'), 'user', User::class)->withMeta([
                'belongsToId' => auth()->user()->id,
            ])->hide(),
            Text::make(__('comment.text'), 'text')->rules(['required', 'max:191']),
        ];
    }
```

We had nova with v4.4.x which was updated to the latest and assets published with nova:publish, after that it is not possible to create comments, because validation fails.
I started to downgrade and got it working only with v4.5.2, so starting with v4.5.3 it is broken.
When it is working, payload looks like this
```
commentable: 58
commentable_type: parentResource
commentable_trashed: false
user: 1
user_trashed: false
text: test
viaResource: parentResource
viaResourceId: 58
viaRelationship: comments
```
When not working:
```
text: test
viaResource: parentResource
viaResourceId: 58
viaRelationship: comments
```

**Comment by crynobone** [comment, marked as answer]
Starting with 4.5.3 we no longer sending field value when it's hidden. Please make it visible or use `Hidden` field if you want it to be hidden from display.

The `show()` and `hide()` method was added specifically for Dependent Field, and when it is set to hidden, it shouldn't send any value.
